src/components/organisms/DashboardPane/DashboardFilteredMenu.tsx

```diff
@@ -0,0 +1,75 @@
+import {useEffect, useMemo, useState} from 'react';
+
+import {useAppSelector} from '@redux/hooks';
+
+import {DashboardMenu} from '@shared/models/dashboard';
+
+import {MenuItem} from './MenuItem';
+
+type IProps = {
+  filterText: string;
+};
+
+const DashboardFilteredMenu: React.FC<IProps> = props => {
+  const {filterText} = props;
+
+  const menuList = useAppSelector(state => state.dashboard.ui.menuList);
+
+  const [activeMenuItemRef, setActiveMenuItemRef] = useState<HTMLElement>();
+
+  const filteredMenu = useMemo(() => {
+    if (!filterText) {
+      return menuList;
+    }
+
+    return menuList
+      .map((menuItem: DashboardMenu) => ({
+        ...menuItem,
+        children: menuItem.children?.filter((m: DashboardMenu) =>
+          m.label.toLowerCase().includes(filterText.toLowerCase())
+        ),
+      }))
+      .filter((menuItem: DashboardMenu) => menuItem.children && menuItem.children?.length > 0)
+      .filter(
+        (menuItem: DashboardMenu) =>
+          menuItem.children &&
+          menuItem.children?.reduce(
+            (total: number, m: DashboardMenu) => total + (m.resourceCount ? m.resourceCount : 0),
+            0
+          ) > 0
+      );
+  }, [filterText, menuList]);
+
+  useEffect(() => {
```

Comment:
do you want this to be triggered only on mount or whenever the `activeMenuItem` changes?
because using the ref as a dep here will trigger this only once on mount